### DISCUSSION TITLE
anycable Golang client

### DISCUSSION THREAD

**Discussion by cinemast** [anycable/anycable#240] [Q&A]
I am having difficulties creating a golang anycable client application that can subscribe and publish to channels. Am I overlooking a client package in this repo?

Do you have any sample snippets on how to publish/subscribe to anycable server from a golang client application?

**Comment by palkan** [comment]
Hey!

Publishing can be done via raw HTTP. It's just a POST request:

```
curl -X POST -H "Content-Type: application/json" -d '{"stream":"my_stream","data":"{\"text\":\"Hello, world!\"}"}' http://localhost:8090/_broadcast
```

As for subscribing, we don't have any official clients yet, but there are some options:

- Use Server-sent events (there should be existing Go implementations). This way you can subscribe to a single channel with just streaming HTTP

- Check out our k6 plugin, which has a client implementation: https://github.com/anycable/xk6-cable/blob/master/client.go

**Reply by cinemast** [reply]
Thank you, I will try that. Is there some sort of API description?

**Reply by cinemast** [reply]
I quickly hacked together one, maybe someone else finds it useful:

https://github.com/cinemast/anycable-go-client

Cheers from the people at [LiveVoice](https://github.com/LiveVoiceOfficial/) 😄
